sample-tracking-status-overview-domain/src/test/groovy/life/qbic/business/samples/count/CountSamplesSpec.groovy

```diff
@@ -104,4 +104,51 @@ class CountSamplesSpec extends Specification {
         1 * output.failedExecution(_)
         0 * output.countedReceivedSamples(_)
     }
+
+    def "successful execution of counting failed qc samples lead to success notifications"() {
+        given:
+        CountSamplesDataSource dataSource = Stub()
+        String projectCode = "QABCD"
+        dataSource.fetchSampleStatusesForProject(projectCode) >> { new ArrayList<Status>() }
+        CountSamplesOutput output = Mock()
+        CountSamples countSamples = new CountSamples(dataSource, output)
+        when:"the use case is run"
+        countSamples.countQcFailedSamples(projectCode)
+        then:"a successful message is send"
+        1 * output.countedFailedQcSamples(projectCode, _ as Integer, _ as Integer)
+        0 * output.failedExecution(_ as String)
+    }
+
+    def "amount of samples failing QC is correctly returned"() {
+        given:
+        CountSamplesDataSource dataSource = Stub()
+        List<Status> statuses = [Status.SAMPLE_QC_FAIL, Status.DATA_AVAILABLE, Status.SAMPLE_QC_FAIL]
+        String projectCode = "QABCD"
+        dataSource.fetchSampleStatusesForProject(projectCode) >> { statuses }
+        CountSamplesOutput output = Mock()
+        CountSamples countSamples = new CountSamples(dataSource, output)
+        when:"the use case is run"
+        countSamples.countQcFailedSamples(projectCode)
+        then:"the correct amounts of samples are returned"
+        1 * output.countedFailedQcSamples(projectCode, 3, 2)
+        0 * output.failedExecution(_ as String)
+    }
+
+
+    def "unsuccessful execution of counting failed qc samples leads to failure notifications"() {
+        given:
+        String projectCode = "QABCD"
+        CountSamplesDataSource dataSource = Stub()
+        dataSource.fetchSampleStatusesForProject(projectCode) >> {
+            throw new RuntimeException("Testing runtime exceptions")
+        }
+        CountSamplesOutput output = Mock()
+        CountSamples countSamples = new CountSamples(dataSource, output)
+        when:"the use case is run"
+        countSamples.countQcFailedSamples(projectCode)
+        then:"a failure message is send"
+        1 * output.failedExecution(_)
+        0 * output.countedFailedQcSamples(_)
+    }
+
```

Comment:
I adapted the test to also cover it, because data available is the latest possible status
![Screenshot 2021-08-06 at 10 25 51](https://user-images.githubusercontent.com/20778777/128481005-b2da481f-eeb0-4dd8-8617-6fc28fce58c1.png)

Reply:
This test does not test for the same